packages/lib-classifier/src/plugins/tasks/SurveyTask/models/SurveyAnnotation.js

```diff
@@ -0,0 +1,21 @@
+import { types } from 'mobx-state-tree'
+import Annotation from '../../models/Annotation'
+
+const SurveyChoice = types.model('SurveyChoice', {
+  answers: types.optional(types.frozen({}), {}),
```

Comment:
Not happy with using `types.frozen` for `answers` or `filters` (below), when

structure for `answers` is a dynamic key with a value of string or array of strings and 

structure for `filters` is a dynamic key with a value of string

but wasn't able to implement using MST types, any help/examples appreciated.

Reply:
I've got an example of a custom type, using `types.refinement`, in #1885. That's validating an array of arrays, to check the type of each array member, but the technique might be applicable to validating an object's values too.

If the keys for `answers` and `filters` are defined by the task, then it might be possible to get the available values from the annotation's task model, then validate the object keys against those. I'm not sure if that would work in practice. I can't remember if an annotation has access to its parent task. I'm also not sure if that level of validation is required. For example, I don't think we validate question task annotations to check that the answer index is a valid answer.

Reply:
This came up for Alice as well. `types.refinement` also used in a case where an object had a variable key name in it. Check out the model: https://github.com/zooniverse/text-editor/blob/e027f8189d605d1e2d7dbadb93a805935e9418b1/src/store/TranscriptionsStore.js#L19-L21

Also, I'm not sure if necessary either. `types.frozen` should be ok for now and we can refine later if necessary.